src/Chainweb/Pact/Backend/ChainwebPactDb.hs

```diff
@@ -0,0 +1,375 @@
+
+_dirtSimpleDbTest :: IO (Either Error ())
+_dirtSimpleDbTest = withTempSQLiteConnection $ \case
+   Left l -> return (Left $ fst l)
+   Right c -> do
+     createBlockHistoryTable c
+     createVersionHistoryTable c
+     createVersionedTablesTable c
+     _checkTableExist c "BlockHistory" >>= print
+     _checkTableExist c "VersionHistory" >>= print
+     _checkTableExist c "VersionedTables" >>= print
+     return (Right ())
+
+data NonVersionInsert
+  = BlockHistory BlockHeight
+                 BlockHash
+  | VersionHistory BlockVersion
+  | VersionedTables TableName BlockVersion
+
+nonVersionInsert  :: Database -> NonVersionInsert -> IO ()
+nonVersionInsert c ins =
+  case ins of
+    BlockHistory bh hsh -> do
+      let s = "INSERT INTO BlockHistory ('blockheight','hash') VALUES (?,?);"
+      exec' c s [SInt (fromIntegral bh), SBlob (encode hsh)]
+    VersionHistory (BlockVersion bh version) -> do
+      let s =
+            "INSERT INTO VersionHistory ('version','blockheight') VALUES (?,?);"
+      exec'
+        c
+        s
+        [SInt (_getReOrgVersion version), SInt (fromIntegral bh)]
+    VersionedTables (TableName name) (BlockVersion bh version) -> do
+      let s =
+            "INSERT INTO VersionedTables ('tableid','blockheight','version') VALUES (?,?,?);"
+      exec'
+        c
+        s
+        [ SText (Utf8 (toS name))
+        , SInt (fromIntegral bh)
+        , SInt (_getReOrgVersion version)
+        ]
+
+assertEqual :: (Eq a, Show a) => String -> a -> a -> IO ()
+assertEqual msg a b = if a == b
+  then return ()
+  else error $ msg ++ ": " ++ show a ++ " is not equal to " ++ show b
+
+_insertionTest  :: IO (Either Error ())
+_insertionTest = withTempSQLiteConnection $ \case
+  Left l -> return (Left $ fst l)
+  Right c -> do
+    createBlockHistoryTable c
+    nonVersionInsert c (BlockHistory (BlockHeight 0) nullBlockHash)
+    res1 <- qry_ c "SELECT * FROM BlockHistory;" [RInt, RBlob]
+    assertEqual "Read the BlockHistoryTable Table" res1 ans1
+    createVersionHistoryTable c
+    nonVersionInsert c (VersionHistory (BlockVersion (BlockHeight 0) (ReorgVersion 0)))
+    res2 <- qry_ c "SELECT * FROM VersionHistory;" [RInt, RInt]
+    assertEqual "Read the VersionHistoryTable Table" res2 ans2
+    createVersionedTablesTable c
+    nonVersionInsert c
+        (VersionedTables (TableName "user1") (BlockVersion (BlockHeight 0) (ReorgVersion 0)))
+    res3 <- qry_ c "SELECT * FROM VersionedTables;" [RText, RInt, RInt]
+    assertEqual "Read the VersionedTables Table" res3 ans3
+    return $ Right ()
+  where
+    ans1 =
+      [ [ SInt 0
+        , SBlob
+            "\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL"
+        ]
+      ]
+    ans2 = [[SInt 0,SInt 0]]
+    ans3 = [[SText "user1",SInt 0,SInt 0]]
+
+
+createBlockHistoryTable :: Database -> IO ()
+createBlockHistoryTable c =
+  exec_
+    c
```

Comment:
This is excessive linebreaking. Please go to "exec_ c \n ..."

Reply:
Fixed.